grpc/stream.go

```diff
@@ -348,24 +351,24 @@ func (s *stream) closeWithError(err error) error {
 	if s.timeoutCancel != nil {
 		s.timeoutCancel()
 	}
-
-	s.logger.WithError(err).Debug("connection closed")
-
-	if err != nil {
-		if errList := s.callErrorListeners(err); errList != nil {
-			return errList
-		}
-	}
-
-	return nil
 }
 
 func (s *stream) callErrorListeners(e error) error {
+	if e == nil {
+		return nil
+	}
+
 	rt := s.vu.Runtime()
 
 	obj := extractError(e)
 
-	for _, errorListener := range s.eventListeners.all(eventError) {
+	list := s.eventListeners.all(eventError)
+
+	if len(list) == 0 {
+		s.logger.Errorf("no handlers for error registered, but an error happened: %s", e)
```

Comment:
Should it be a warning? If no handler is set then probably they don't care. :shrug:

Reply:
I believe it's still better to use the error severity because there are no handlers. were set up by mistake, but if an error happens there, it's something that requires attention

Reply:
@mstoykov, what's your opinion here? :relaxed:

Reply:
I do agree this is an error. But arguably the log level of `error` is used for stuff that need to stop the execution. 

So I feel like warning will be a better fit.

Arguably an error should mean that something is definitely wrong. And no error handling while ... not good idea - is not necessarily meaning "your script does not work correctly". 

There really isn't much the user can do in an error handler than .... log it and potentially do some clean up.

Reply:
Okay, changed to the `Warnf` :+1: